### ISSUE TITLE
AuthServer built with Release configuration always responds with wrong password

### ISSUE THREAD

**Issue by Malow** [azerothcore/azerothcore-wotlk#2358]
##### SMALL DESCRIPTION:
Building the AuthServer with Release configuration makes it always respond with wrong password when someone tries to log in. It works perfectly when built with Debug configuration. I tried both test1/a, ".account set password test1 1" and then test1/1, and creating a completely new account. All 3 fails. 

##### STEPS TO REPRODUCE THE PROBLEM:
1. Build latest AC in Release-mode
2. Run AuthServer.exe
3. Try to log on with any account using the client.

##### EXTRA NOTES:
While I was doing some binary search testing of commits for another issue to find the culprit commit for that issue I did for this issue simultaneously. I've narrowed it down to 0ee3f8373a04b7c654040cde6f45cfc6bfd0b0d2 working fine, and b22a1fd752899be56823658546db4ce5b31661b1 not working. If I had to guess the likely culprit is 3a43e8c415cee4041f2a59e9ebc8af721a49be23 (@FrancescoBorzi, @Winfidonarleyan). Now I'm not sure if there's anything new I need to be doing with this commit compared to older commits to get it working again, if so then please advice what that is :)

##### BRANCH(ES):
master

##### AC HASH/COMMIT:
003e8fa4598bcbbc28367b752c1183fed84a72f8

##### OPERATING SYSTEM:
Windows 10, latest updates.

##### MODULES:
None, able to reproduce on completely fresh AC.

##### OTHER CUSTOMIZATIONS:
None, able to reproduce on completely fresh AC.

**Comment by Stoabrogga** [comment]
I'm currently on commit 7d09a260e8a8fde8074a4af89ab636d026384617 and always use "Release" for my server. I'm not facing any issues so far (Ubuntu 16.04; compiled using cmake 3.15.4 and clang 7.1.0; clean DB).

**Comment by LichKing255** [comment]
me too don't have problem.
Win 10 last update.

**Comment by Malow** [comment]
So the only thing that I can see that I'm doing different from the installation-guide is that I'm not using CMAKE_INSTALL_PREFIX, instead manually moving the compiled files into the folder I want to run them from. Could not using CMAKE_INSTALL_PREFIX be the reason why I'm experiencing this issue?

**Comment by Stoabrogga** [comment]
I think you have to specify the install prefix or otherwise the server looks for the config files in a default directory. If I remove this parameter the default seems to be /usr/local on Linux:
```
* Install core to                 : /usr/local
* Install libraries to            : /usr/local/lib
* Install configs to              : /usr/local/etc
```

**Comment by Malow** [comment]
My AuthServer is able to correctly connect to my DB, so I don't think there's any problems with my config files not being found:
![image](https://user-images.githubusercontent.com/949807/67002527-6507eb80-f0dc-11e9-8b7a-2374093e069c.png)

**Comment by Stoabrogga** [comment]
You could test using the install prefix parameter and see if that works.

**Comment by Malow** [comment]
I could've sworn that the first thing I did was to try to update my Visual Studio version, but it seems not because when I checked for updates now there was one available, and upgrading to it solved this issue. So odd that the older version did not give a compilation error nor a run-time error, just that the actual program didn't function correctly in Release mode. 

So since commit 3a43e8c415cee4041f2a59e9ebc8af721a49be23 Version 15.9.17+ of Microsoft Visual Studio 2017 should be used.
